### PR TITLE
CircularOptionPicker: stop using composite store

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Requires #64832 to be merged first
Extracted from #64723

Refactor `CircularOptionPicker` so that it doesn't use the `store` from `useCompositeStore` to function.

## Why?

See https://github.com/WordPress/gutenberg/issues/63704#issuecomment-2305291168

## How?

By controlling the `activeId` state via props.

## Testing Instructions

Make sure that existing usages of `CircularOptionPicker` continue to work as expected.

In particular, make sure that, when `CircularOptionPicker` has a selected option (and the selected option is not the first option), tabbing on the component for the first time will automatically focus the selected option, instead of the first option.

<details>

<summary>This can be tested in Storybook by applying the following diff — in the default example, the second item (green) should now be initially selected </summary>

```diff
diff --git a/packages/components/src/circular-option-picker/stories/index.story.tsx b/packages/components/src/circular-option-picker/stories/index.story.tsx
index e091a2ac54..e2eb2b6cd5 100644
--- a/packages/components/src/circular-option-picker/stories/index.story.tsx
+++ b/packages/components/src/circular-option-picker/stories/index.story.tsx
@@ -46,7 +46,9 @@ const meta: Meta< typeof CircularOptionPicker > = {
 	decorators: [
 		// Share current color state between main component, `actions` and `options`
 		( Story ) => {
-			const [ currentColor, setCurrentColor ] = useState< string >();
+			const [ currentColor, setCurrentColor ] = useState<
+				string | undefined
+			>( '#0f0' );
 
 			return (
 				<CircularOptionPickerStoryContext.Provider

```
</details>

### CODE REVIEW

File: packages/components/src/circular-option-picker/circular-option-picker-option.tsx [hunk 1]
```diff
@@ -46,18 +45,21 @@ function UnforwardedOptionAsOption(
...
-		compositeStore.setActiveId( id );
-	}
+	const { setActiveId, activeId } = useContext( CircularOptionPickerContext );
+
+	useEffect( () => {
+		if ( isSelected && ! activeId ) {
+			// The setTimeout call is necessary to make sure that this update
+			// doesn't get overridden by `Composite`'s internal logic, which picks
+			// an initial active item if one is not specifically set.
+			window.setTimeout( () => setActiveId?.( id ), 0 );
```
Comment (ciampo) [inline packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: https://github.com/WordPress/gutenberg/pull/64723#discussion_r1735082995

Comment (ciampo) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: Hey @diegohaz , do you have any advice better alternatives to using `setTimeout` here? Without it, it looks like `Composite`'s internal logic to pick an initially selected item overrides this selection.

@tyxla suggested using `queueMicrotask`, while @mirka suggesting calling `setItems` instead. (see link above). What's your take? Is this behaviour expected? Is there a more elegant way to make this work?

Comment (diegohaz) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: After reading the comment and the discussion you linked, I still don't understand its purpose. Could you clarify?

Does Ariakit still set an active id even after you set one yourself?

Comment (ciampo) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: `CircularOptionPicker` displays a few color options as a composite widget.

If the component loads and then a color option is selected (programmatically), then we'd like to set that color option as the active composite item, so that when a user moves its focus on the composite widget, the selected color (and not the first color in the list) gets focus.

Previously, the component was calling `compositeStore.setActiveId( id )` insider the render function (not wrapped in `useEffect`), and that seemed to work. 

With the changes from this PR, though, our call to `setActiveId` gets almost immediately overridden by some internal ariakit logic which sets the active item to the first item _after_ out `setActiveId` call.

> Does Ariakit still set an active id even after you set one yourself?

In short, it seems so. That's why we're discussing about the timing of the call.

Comment (ciampo) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: An alternative that came to mind — would it be better to set the active id in `onFocusVisible` ?

File: packages/block-editor/src/components/color-palette/test/control.js [hunk 2]
```diff
@@ -10,9 +10,22 @@ import ColorPaletteControl from '../control';
 
 const noop = () => {};
 
+async function renderAndValidate( ...renderArgs ) {
```
Comment (ciampo) [inline packages/block-editor/src/components/color-palette/test/control.js:13]: Not sure what we ended up deciding re. the `render` function — for now, I'm using a composite-specific workaround, but I'm not sure how well this would scale

File: packages/components/src/circular-option-picker/circular-option-picker-option.tsx (hunk 1 above)
Comment (ciampo) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: > An alternative that came to mind — would it be better to set the active id in `onFocusVisible` ?

I looked into it, and it would make the code even more convoluted, since it would cause the composite wrapper to also become focusable.

What we need is:
- if the user hasn't interacted yet with the composite widget AND there is a selected option, that option should become the active item — so that focus moves to that option when the user first focuses the widget;
- in any other scenario, the first item should be the active item

@diegohaz hope the context is a bit more clear around what we're trying to achieve here

File: packages/block-editor/src/components/color-palette/test/control.js (hunk 2 above)
Comment (tyxla) [reply packages/block-editor/src/components/color-palette/test/control.js:13]: Looks good for this particular instance. Since this is a testing detail, let's worry about it if it actually needs to scale.

File: packages/components/src/circular-option-picker/circular-option-picker-option.tsx (hunk 1 above)
Comment (tyxla) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: I think this looks good to go. We can always iterate if @diegohaz has a better suggestion.

**Review by tyxla — APPROVED** [review]
This tests well and the code looks good 👍 

If we need to adjust the active ID logic, we can always do it in a subsequent PR.

Thanks 🚀

File: packages/components/src/circular-option-picker/circular-option-picker-option.tsx (hunk 1 above)
Comment (diegohaz) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: I tried to reproduce this with Ariakit, but I might be missing something. There's no need for `setTimeout` here: https://stackblitz.com/edit/cnsby2?file=command%2Findex.tsx&theme=dark

Can you try to reproduce it from that code? This could be a bug.

Edit: I can reproduce it if I pass down an external `setActiveId` instead of calling `setActiveId` from the composite store: https://stackblitz.com/edit/cnsby2-kp1gfx?file=command%2Findex.tsx&theme=dark

In this case, a better workaround is calling `setActiveId` from the Ariakit composite store if you have access to it.

https://github.com/ariakit/ariakit/issues/4100

Comment (ciampo) [reply packages/components/src/circular-option-picker/circular-option-picker-option.tsx:60]: Thank you for looking into it!

> a better workaround is calling `setActiveId` from the Ariakit composite store if you have access to it.

The main reason for this refactor is to avoid using the store directly from outside the `Composite` component, therefore we're probably going to wait for https://github.com/ariakit/ariakit/issues/4100
